docs/node2node.md

```diff
@@ -0,0 +1,107 @@
+# node to node messages
+
+
+[summary placeholder]
+
+(todo: fully describe pop and shift mechanism)
+(todo: calcuate optimal data sizes)
+
+
+## streamed requests
+
+direct node to node request/response messages sent on a bidi quic stream.
+
+a stream carries 1 type of request/response messages. we maintain 1 stream per type to each remote node with the exception of path stream which requires multiple instances of itself.
+
+the stream starts by sending 2 bytes identifying the stream kind of which there are the following:
+
+* network data lookup
+* network data upsert
+* path build
+* path update
+* path stream
+
+### network data lookup stream
+
+used for fetching of:
+
+* pubkey list with unlock times if they are applicable.
+* pubkey list delta given last hash of full list.
```

Comment:
The purpose of these two endpoints is not obvious to me.  (Not saying there isn't one, just that it isn't obvious from just this description).